### ISSUE TITLE
caffe test should automatically run over whole test set

### ISSUE THREAD

**Issue by kamikawa** [BVLC/caffe#1690]
Hi 

I am beginning to use Caffe, and feel a little inconvenient about "caffe test" command spec.
Usually, test sample accuracy would be calculated by whole test samples, but "caffe test" command calculate only "batch size" x "iterations".

My idea is that "caffe test" model uses deploy.prototxt style and inside caffe command, number of test samples is counted automatically, and  output score for whole samples.

Best regards.

**Comment by lukeyeager** [comment]
It would be nice if this was done in training, too. Other frameworks use the "epoch" idea. So you would train on 10 epochs of the training data, and then test 1 epoch of the validation data, for example.

I mean sure, I can calculate what the iterations should be for one epoch based on the batch size myself, but why do you make me do that? Is there an explanation for caffe's use of "iterations" somewhere?

**Comment by shelhamer** [comment]
Closing as duplicate of #1094. This is a convenience, but not strictly necessary.
